### PR TITLE
chore: replace `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`

### PR DESCRIPTION
**Pull request by Juneezee**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

The `github.com/ghodss/yaml` package is no longer actively maintained. There are numerous inquiries about the project's status on its issue tracker: https://github.com/ghodss/yaml/issues. `sigs.k8s.io/yaml` is a permanent fork of `github.com/ghodss/yaml`, which is actively maintained by Kubernetes SIG and widely used in K8s projects.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:
```release-note
chore: replace unmaintained `github.com/ghodss/yaml` dependency with `sigs.k8s.io/yaml`
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -21,6 +20,7 @@ require (
 	k8s.io/pod-security-admission v0.0.0
 	k8s.io/utils v0.0.0-20230209194617-a36077c30491
 	sigs.k8s.io/sig-storage-lib-external-provisioner/v6 v6.3.0
+	sigs.k8s.io/yaml v1.3.0
```
Comment (msau42) [inline go.mod:23]: Does vendor need to be updated with the new package?

Comment (Juneezee) [reply go.mod:23]: It was actually an indirect dependency before this pull request was made. Therefore, it has already been included in the vendor directory. https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/tree/56a9cf7443ccc10359569ca229a10035205d4e26/vendor/sigs.k8s.io/yaml

https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/56a9cf7443ccc10359569ca229a10035205d4e26/go.mod#L133

**Comment by mauriciopoppe** [comment]
Thank you for your contribution!

/lgtm
/approve

**Pull request merged**